### PR TITLE
Readability improvements to the ECS resource detection processor

### PR DESCRIPTION
**Pull request by rakyll**
Simplified abstractions and tried to rely on what's in the standard library instead of introducing new interfaces.

### CODE REVIEW

**Comment by bogdandrutu** [comment]
Please rebase

**Comment by rakyll** [comment]
PTAL, I think the failure is coming from the code coverage.

File: processor/resourcedetectionprocessor/internal/aws/ecs/metadata.go [hunk 1]
```diff
@@ -14,13 +14,94 @@
...
-	Do(req *http.Request) (*http.Response, error)
+type metadataProvider interface {
+	fetchTask(tmde string) (*TaskMetaData, error)
+	fetchContainer(tmde string) (*Container, error)
 }
 
-type ecsMetadataProvider interface {
-	fetchTaskMetaData(tmde string) (*TaskMetaData, error)
-	fetchContainerMetaData(tmde string) (*Container, error)
+type TaskMetaData struct {
```
Comment (bogdandrutu) [inline processor/resourcedetectionprocessor/internal/aws/ecs/metadata.go:28]: s/TaskMetaData/TaskMetadata? For consistency with other places where you use "Metadata" as one word?

File: processor/resourcedetectionprocessor/internal/aws/ecs/metadata.go [hunk 2]
```diff
@@ -14,13 +14,94 @@
...
+		tmdeResp = &TaskMetaData{}
+	} else {
+		tmdeResp = &Container{}
+	}
+
+	err = json.NewDecoder(resp.Body).Decode(tmdeResp)
+	defer resp.Body.Close()
+
+	if err != nil {
+		log.Printf("Encountered unexpected error reading response from ECS Task Metadata Endpoint: %v", err)
```
Comment (jrcamp) [inline processor/resourcedetectionprocessor/internal/aws/ecs/metadata.go:102]: You'll want to use zap log instance passed in now.

File: processor/resourcedetectionprocessor/internal/aws/ecs/metadata.go (hunk 1 above)
Comment (rakyll) [reply processor/resourcedetectionprocessor/internal/aws/ecs/metadata.go]: Done.

File: processor/resourcedetectionprocessor/internal/aws/ecs/metadata.go (hunk 2 above)
Comment (rakyll) [reply processor/resourcedetectionprocessor/internal/aws/ecs/metadata.go:103]: Can send it in another PR. This PR is just improving some readability issues. Added a TODO.

**Comment by bogdandrutu** [comment]
Please rebase, the zap logger already changed I think :)

**Comment by bogdandrutu** [comment]
Please fix lint errors:
```

make[1]: Entering directory '/home/circleci/project/processor/routingprocessor'
internal/aws/ecs/ecs_test.go:22: File is not `gofmt`-ed with `-s` (gofmt)
	"go.uber.org/zap"
internal/aws/ecs/metadata_test.go:24: File is not `gofmt`-ed with `-s` (gofmt)
	"go.uber.org/zap"
internal/aws/ecs/ecs_test.go:25: File is not `goimports`-ed with -local github.com/open-telemetry/opentelemetry-collector-contrib (goimports)
	"go.opentelemetry.io/collector/component"
make[1]: *** [../../Makefile.Common:115: lint] Error 1
```

**Pull request closed without merging**